File: 219-contains-duplicate-ii/contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        int l = 0;

        HashSet<Integer> window = new HashSet<>();

        for( int r = 0 ; r < nums.length ; r++)
        {
            if( r - l > k)
            {
                window.remove(nums[l]);
                l += 1;
            }
            if (window.contains(nums[r]))
            {
                return true;
            }
            window.add(nums[r]);
        }

        return false;
    }
}

File: 219-contains-duplicate-ii/contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        l = 0 
        r = 0 
        window = set()

        while r < len(nums):

            if r - l > k:
                window.remove(nums[l])
                l += 1

            if nums[r] in window:
                return True

            window.add(nums[r]) 
            r += 1
            
        return False


        
        